docs/source/format/how_manager.rst

```diff
@@ -0,0 +1,204 @@
+
+   Now, the application asks the driver for a table of functions to call.
+
+Then, the application uses the driver by calling the functions in the table:
+
+.. figure:: DriverTableUse.mmd.svg
+
+   The application uses the table to call driver functions.  This approach
+   scales to multiple drivers.
+
+Dealing with the table, however, is messy.  So the overall recommended
+approach is to use the ADBC driver manager.  This is a library that pretends
+to be a single driver that can be linked to and used "like normal".
+Internally, it loads the table of function pointers and tracks which
+database/connection/statement objects need which "actual" driver, making it
+easy to dynamically load drivers at runtime and use multiple drivers from the
+same application:
+
+.. figure:: DriverManagerUse.mmd.svg
+
+   The application uses driver manager to "feel like" it's just using a single
+   driver.  The driver manager handles the details behind the scenes.
+
+In More Detail
+==============
+
+The adbc.h header ties everything together.  It is the abstract API
+definition, akin to interface/trait/protocol definitions in other languages.
+C being C, however, all it consists of is a bunch of function prototypes and
+struct definitions without any implementation.
+
+A driver, at its core, is just a library that implements those function
+prototypes in adbc.h.  Those functions may be implemented in C, or they can be
+implemented in a different language and exported through language-specific FFI
+mechanisms.  For example, the Go and C# implementations of ADBC can both
+export drivers to consumers who expect the C API definitions.  As long as the
+definitions in adbc.h are implemented somehow, then the application is
+generally none the wiser when it comes to what's actually underneath.
+
+How does an application call these functions, though?  Here, there are several
+options.
+
+Again, the simplest case is as follows: if (1) the application links directly
+to the driver, and (2) the driver exposes the ADBC functions *under the same
+name* as in adbc.h, then the application can just ``#include <adbc.h>`` and
+call ``AdbcStatementExecuteQuery(...)`` directly.  Here, the application and
+driver have a relationship no different than any other C library.
+
+.. figure:: DriverDirectLink.mmd.svg
+
+   In the simplest case, an application directly links to the driver and calls
+   ADBC functions.  When the application calls ``StatementExecuteQuery``, that
+   is directly provided by the driver that it links against.
+
+Unfortunately, this doesn't work as well in other scenarios.  For example, if
+an application wishes to use multiple ADBC drivers, this no longer works: both
+drivers define the same functions (the ones in adbc.h), and when the
+application links both of them, the linker has no way of telling which
+driver's function is meant when the application calls an ADBC function.  On
+top of that, this violates the `One Definition Rule`_.
```

Comment:
If two drivers try to expose both the `Adbc....` methods and their own aliases, won't we still get a duplicate symbol issue when dynamically loading both? Should we indicate that the symbols need to be weak symbols to prevent that?

Reply:
I don't think that's an issue at this point (well, I suppose it's an issue if you still try and use the symbols), but we don't mark any of our symbols as weak.

Reply:
It might end up being an issue in the future that we should keep in mind then at least.

Reply:
Can you file a separate issue then?

Reply:
will do